circuit-construction/src/writer.rs

```diff
@@ -546,17 +636,22 @@ pub trait Cs<F: FftField + PrimeField> {
         let one = F::one();
         let neg_one = -one;
 
+        // For each of the chunks, get the corresponding bits
         for (i, row_bits) in bits_msb[..].chunks(bits_per_row).enumerate() {
             let mut row: [Var<F>; COLUMNS] = array_init(|_| self.var(|| F::zero()));
             row[0] = n;
             row[2] = a;
             row[3] = b;
 
+            // For this row, get crumbs of 2 bits each
             for (j, crumb_bits) in row_bits.chunks(2).enumerate() {
-                let b0 = crumb_bits[1];
-                let b1 = crumb_bits[0];
+                // Remember the MSB of each crumb is in the 0 index
+                let b0 = crumb_bits[1]; // less valued
+                let b1 = crumb_bits[0]; // more valued
 
+                // Value of the 2-bit crumb in MSB
                 let crumb = self.var(|| b0.val() + b1.val().double());
+                // Stores the 8 of them in positions [6..13] of the row
                 row[6 + j] = crumb;
 
                 a = self.var(|| {
```

Comment:
Here, I don't see why this is useful. There are no 4 different branches for the 4 different values of the crumb, but instead just 3. And again, we don't use `a` nor `b` later, so this code seems to have no effects later.